js/PanoViewer.js

```diff
@@ -0,0 +1,379 @@
+            const viewLimiter = Marzipano.RectilinearView.limit.traditional(
+                config.MAX_RESOLUTION,
+                degree_to_radian * config.MAX_FOV
+            );
+
+            view = new Marzipano.RectilinearView({}, viewLimiter);
+            console.log(view);
+            scene = viewer.createScene({
+                source: source,
+                geometry: new Marzipano.CubeGeometry(config.LEVEL_PROPERTIES_LIST),
+                view: view,
+                pinFirstLevel: true
+            });
+
+            self.addHotspots(hotspots, scene);
+            view.setYaw(degree_to_radian * heading);
+            view.setPitch(degree_to_radian * pitch);
+            view.setFov(degree_to_radian * fov);
+
+            scene.switchTo();
+
+            // Adding event handling
+            viewer.controls().addEventListener('active', function(evt, elem, bla) {
+                const parameters = view.parameters();
+                console.log('active', parameters);
+            });
+            viewer.controls().addEventListener('inactive', function(evt, elem, bla) {
+                console.log('inactive', evt, elem, bla);
+            });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    /**
+     * Initialize the Marzipano viewer.
+     * init is used to determine the insertion point
+     */
+    function init (insertion) {
+        const panoElement = self.getElement(insertion);
+        if (!panoElement) {
+            // @TODO Error handling
+            return 'ERROR!';
+        }
+        viewer = new Marzipano.Viewer(panoElement, viewerOpts);
+    };
+
+    /**
+     * 
+     * @param {float} lat - Latitude 
+     * @param {float} lon - Longtitude
+     * @param {int} cappedRadius - Max search radius (optional)
+     * @param {float} heading - initial heading (optional)
+     * @param {float} pitch - initial pitch (optional)
+     * @param {float} fov - Field of vision (optional)
+     * 
+     * This is the main api interaction for loading the panorama view.
+     * After this has been called navigating within the panorama via hotspots
+     * is handled by FUNCTION_NAME
+     */
+    function loadPanorama (lat, lon, cappedRadius, heading, pitch, fov) {
+        const url = config.API_ROOT + config.PANO_URI + '?lat=' +
+                lat+ '&lon=' + lon + '&radius=' + config.RADIUS;
+
+        // Setting defaults if not given
+        fov = fov || config.FOV;
+        heading = heading || 0;
+        pitch = pitch || 0;
+
+        return (self.getPanoramaData(url))
+            .then(loadScene.bind(null, heading, pitch, fov));
+    };
+
+    const updatePanorama = function (pano_id) {
+        const url = config.API_ROOT + config.PANO_URI + pano_id;
+
+        const fov = 80,
+            heading = 0,
+            pitch = 0;
+        return (self.getPanoramaData(url))
+            .then(loadScene.bind(null, heading, pitch, fov));
+    };
+
+    function updateConfig (opts) {
```

Comment:
Deze functie zou ik simpelweg nog even weglaten. Custom config kan ook bij de init methode mee worden gestuurd. Dat is naar mijn idee logischer dan achteraf wijzigen